### PR TITLE
Schedule

### PR DESCRIPTION
**Pull request by jesseli2002**
Incorporate rocket scheduler.

### CODE REVIEW

**Comment by jesseli2002** [comment]
hmm, maybe something with git submodules that needs to be dealt with

**Comment by jesseli2002** [comment]
[Organize code to more generally run tasks at different intervals](https://trello.com/c/2YIzfqkl/287-organize-code-to-more-generally-run-tasks-at-different-intervals)

File: avionics/common/include/scheduler_clock.hpp
Comment (akoen) [inline avionics/common/include/scheduler_clock.hpp:5]: RocketSchedule/schedule.hpp gives me a file not found error - does not compile

**Review by akoen — CHANGES_REQUESTED** [review]

File: avionics/common/include/scheduler_clock.hpp
Comment (jesseli2002) [reply avionics/common/include/scheduler_clock.hpp:5]: Run `git submodule update` and try again @SilentSys

Comment (SilentSys) [reply avionics/common/include/scheduler_clock.hpp:5]: @jesseli2002 Looks like you'll probably need to update the Tavis task in the GS to get the submodule https://stackoverflow.com/questions/3796927/how-to-git-clone-including-submodules

Or maybe better yet, you could add `git submodule update` as a build system step to cover the above case as well as ensure that people don't forget to update, build with the wrong version, and then experience a bug/inconsistency.

**Review by SilentSys — CHANGES_REQUESTED** [review]
I think we really need to safeguard against people checking out the latest commit while having an old version of the submodule, building, flashing the MB and expecting things to behave sanely.

Also this will break the GS Travis job as soon as the required_flare.txt is updated due to it not cloning this repo with submodules enabled.

See my other response for more info on both.

File: avionics/common/include/scheduler_clock.hpp
Comment (jesseli2002) [reply avionics/common/include/scheduler_clock.hpp:5]: https://stackoverflow.com/questions/4611512/is-there-a-way-to-make-git-pull-automatically-update-submodules/49427199

Seems like there's a way to get submodules to automatically update, with a git config thing. Travis commands on FLARE incorporate submodules - it's probably best to just copy what they use into the [.travis.yml in FLAREGUN](https://github.com/UBC-Rocket/UBCRocketGroundStation/pull/57)

**Review by SilentSys — APPROVED** [review]
LGTM. We should still wait until CI is back up before we merge though.

**Review by SilentSys — CHANGES_REQUESTED** [review]
Ran the CI checks manually, pio build failed for TantalusStage1:
```
~/FLARE$ pio run
Processing TantalusStage1 (board: teensy36; framework: arduino; platform: teensy)
-------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/teensy/teensy36.html
PLATFORM: Teensy (4.11.1) > Teensy 3.6
HARDWARE: MK66FX1M0 180MHz, 256KB RAM, 1MB Flash
DEBUG: Current (jlink) External (jlink)
PACKAGES: 
 - framework-arduinoteensy 1.153.0 (1.53) 
 - toolchain-gccarmnoneeabi 1.50401.190816 (5.4.1)
LDF: Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 87 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <i2c_t3>
|-- <SD> 1.2.2
|   |-- <SPI> 1.0
|-- <SPI> 1.0
|-- <TinyGPS>
Submodule 'avionics/common/lib/RocketScheduler' (https://github.com/UBC-Rocket/RocketScheduler.git) registered for path 'avionics/common/lib/RocketScheduler'
Cloning into '/home/bionic/FLARE/avionics/common/lib/RocketScheduler'...
Building in release mode
Compiling .pio/build/TantalusStage1/src/common/lib/CalcHelpers/barometric.cpp.o
Compiling .pio/build/TantalusStage1/src/common/src/XBee.cpp.o
Compiling .pio/build/TantalusStage1/src/common/src/avionics.cpp.o
Compiling .pio/build/TantalusStage1/src/common/src/battery.cpp.o
In file included from avionics/rockets/tantalus_stage_1/include/scheduler.hpp:3:0,
                 from avionics/common/src/avionics.cpp:68:
avionics/common/include/scheduler_clock.hpp:5:39: fatal error: RocketSchedule/schedule.hpp: No such file or directory
compilation terminated.
*** [.pio/build/TantalusStage1/src/common/src/avionics.cpp.o] Error 1
========================= [FAILED] Took 5.46 seconds =========================
```

**Review by SilentSys — APPROVED** [review]
AndreiCI passed. Feel free to merge.

**Pull request merged**
